operator/apis/loki/v1/lokistack_types.go

```diff
@@ -563,6 +563,13 @@ type ObjectStorageSchema struct {
 	// +required
 	// +kubebuilder:validation:Required
 	EffectiveDate StorageSchemaEffectiveDate `json:"effectiveDate"`
+
+	// SchemaStatus conveys any warnings/messages that apply
+	// to the currently applied schema
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	SchemaStatus string `json:"schemaStatus,omitempty"`
```

Comment:
The status should not be part of the type used in the `spec`. We currently re-use the same type for both `spec` and `status` but just because there's no difference in the data. This will change that, which means that we need to define a separate type for the `status` side. That type _can_ embed the current `ObjectStorageSchema` type though, so that it has the same attributes which can be initialized from an existing type.

Reply:
You're right, thanks for pointing this out